pkg/cluster/cluster_info_scraper.go

```diff
@@ -318,30 +324,33 @@ func (s *ClusterScraper) GetPodGrandparentInfo(pod *api.Pod) (string, string, st
 			Resource: commonutil.ReplicationControllerResName}
 	case commonutil.KindReplicaSet:
 		res = schema.GroupVersionResource{
-			Group:    commonutil.K8sAPIReplicasetGV.Group,
-			Version:  commonutil.K8sAPIReplicasetGV.Version,
+			Group:    commonutil.K8sAPIDeploymentGV.Group,
+			Version:  commonutil.K8sAPIDeploymentGV.Version,
 			Resource: commonutil.ReplicaSetResName}
 	default:
 		s.cacheControllerInfo(podControllerInfoKey, kind, name, uid)
-		return kind, name, uid, nil
+		return kind, name, uid, nil, nil, nil
 	}
 
-	obj, err := s.DynamicClient.Resource(res).Namespace(pod.Namespace).Get(name, metav1.GetOptions{})
+	namespacedClient := s.DynamicClient.Resource(res).Namespace(pod.Namespace)
+	obj, err := namespacedClient.Get(name, metav1.GetOptions{})
 	if err != nil {
-		return "", "", "", fmt.Errorf("failed to get %s[%v/%v]: %v", kind, pod.Namespace, name, err)
+		err = fmt.Errorf("Failed to get %s[%v/%v]: %v", kind, pod.Namespace, name, err)
+		glog.Error(err.Error())
+		return "", "", "", nil, nil, err
 	}
 	//2.2 get parent's parent info by parsing ownerReferences:
 	rsOwnerReferences := obj.GetOwnerReferences()
 	if rsOwnerReferences != nil && len(rsOwnerReferences) > 0 {
 		gkind, gname, guid := util.ParseOwnerReferences(rsOwnerReferences)
-		if len(gkind) > 0 && len(gname) > 0 && len(guid) > 0 {
-			s.cacheControllerInfo(podControllerInfoKey, gkind, gname, guid)
-			return gkind, gname, guid, nil
+		if len(gkind) > 0 && len(gname) > 0 {
```

Comment:
Nit: Although it's unlikely `guid` is empty when `gkind` and `gname` exist,  I think it would be safer to keep the previous `&& len(guid) > 0` condition.

Reply:
Updated. I think this too was introduced while rebasing these changes on latest.